src/message_type.rs

```diff
@@ -193,6 +193,23 @@ impl ParsedMessage {
 
         Self::from_peer_message(peer_message, true)
     }
+
+    /// Constructs a copy of this message with the given message type
+    #[allow(needless_pass_by_value)]
```

Comment:
just curious, why is pass by value required here?

Reply:
The overall correct thing to do is going to be deriving `Copy` for `PbftMessageType`, see https://stackoverflow.com/a/44695401 for some rationale on that. However, making that change is a little out-of-scope for this PR, as it's going to involve changing a fair number of lines of code that are currently passing by reference. Passing by value is used here in anticipation of making that change, though.

Reply:
thx